app/controllers/criteria_controller.rb

```diff
@@ -41,6 +41,8 @@ def create
       end
       head :unprocessable_entity
     end
+    @criterion.save!
+    @criterion.set_default_levels if params[:criterion_type] == 'RubricCriterion'
```

Comment:
Don't do this here, move this to inside the `if` statement above. This eliminates the redundant save.